docs/site-generator/Makefile

```diff
@@ -1,155 +1,98 @@
 
-export_filename = $(SHARED)/landing-common.tar
-# Imports common stuff
-import-commons:
-	$(MAKE) -C $(LANDING_RELATIVE_PATH) export-landing-commons target=$(export_filename)
-	$(UNTAR) $(export_filename) --directory $(SITE_TMP_PATH)
-	$(REMOVE) $(export_filename)
-
-repository_filename = repo.tar.gz
-# Downloads the repository whose documentation-site must be generated
-checkout-repository:
-	$(MKDIR) $(REPO_TMP_PATH); \
-	cd $(REPO_TMP_PATH); \
-	$(CURL) $(URL_TARBALL) -o $(repository_filename); \
-	$(UNTARGZ) $(repository_filename) --strip 1;
-	$(REMOVE) $(repository_filename);
-
-hugo_content_raw_path = $(SITE_TMP_PATH)/hugo/content/raw
-project_yml_output = $(SITE_TMP_PATH)/hugo/data/project.yml
-caddyfile_output = $(OUTPUT)/Caddyfile
+hugo_content_raw_path = $(site_sources_path)/hugo/content/raw
+project_yml_path = $(site_sources_path)/hugo/data
 # Extracts the documentation contents from the project repository
 extract-repo-documentation:
-	$(COPY) $(REPO_TMP_PATH)/README.md $(hugo_content_raw_path)
+	$(copy) $(SOURCES)/README.md $(hugo_content_raw_path)
 
-	$(COPY) $(REPO_TMP_PATH)/LICENSE $(hugo_content_raw_path)
+	$(copy) $(SOURCES)/LICENSE $(hugo_content_raw_path)
 
-	@if [[ -d $(REPO_TMP_PATH)/_examples ]]; then \
-		$(COPY) $(REPO_TMP_PATH)/_examples/* $(hugo_content_raw_path)/examples/; \
+	@if [ -d $(SOURCES)/_examples ]; then \
+		$(copy) $(SOURCES)/_examples/* $(hugo_content_raw_path)/examples/; \
 	fi;
 
-	@if [[ -d $(REPO_TMP_PATH)/_tutorials ]]; then \
-		$(COPY) $(REPO_TMP_PATH)/_tutorials/* $(hugo_content_raw_path)/tutorials/; \
+	@if [ -d $(SOURCES)/_tutorials ]; then \
+		$(copy) $(SOURCES)/_tutorials/* $(hugo_content_raw_path)/tutorials/; \
 	fi;
 
-	@if [[ -d $(REPO_TMP_PATH)/_downloads ]]; then \
-		$(COPY) $(REPO_TMP_PATH)/_downloads/* $(hugo_content_raw_path)/downloads/; \
+	@if [ -d $(SOURCES)/_downloads ]; then \
+		$(copy) $(SOURCES)/_downloads/* $(hugo_content_raw_path)/downloads/; \
 	fi;
 
-	@envsubst < $(MAKEFILE_TEMPLATE_PATH)/Caddyfile > $(caddyfile_output)
-
-	@envsubst < $(MAKEFILE_TEMPLATE_PATH)/project.yml > $(project_yml_output)
-
-	@if [ ! -z "$(LANGUAGES)" ]; then \
-		docUrls="docUrls:\n"; \
-		for lang in $(LANGUAGES); do \
-			docUrls+="  - lang: $$lang\n"; \
-			if [ "$$lang" == "go" ]; then \
-				docUrls+="    url: https://godoc.org/github.com/$(OWNER)/$(REPOSITORY)"; \
-			else \
-				docUrls+="    url: $(BASE_URL)/api/$$lang/"; \
-			fi; \
-			docUrls+="\n"; \
-		done; \
-		echo -e "$$docUrls" >> $(project_yml_output); \
-	fi;
+	@envsubst < $(makefile_templates_paths)/project.yml > $(project_yml_path)/project.yml
 
 # Builds the documentation-site
 site-build:
-	$(MAKE) -C $(SITE_TMP_PATH) hugo-build
-	$(COPY) $(SITE_TMP_PATH)/public $(OUTPUT)
+	$(MAKE) -C $(site_sources_path) hugo-build DESTINATION=$(OUTPUT)
 
 # Cleans up
 clean-up:
-	$(REMOVE) $(REPO_TMP_PATH)
-	$(REMOVE) $(SITE_TMP_PATH)
+	$(remove) $(site_sources_path)
+
+# Ensures that configuration is provided
+ensure-building-conf:
```

Comment:
all this is not needed. This is guaranteed to be provided. Is not a makefile for an user but for the building